### PR TITLE
[feature](nereids) add scalar subquery expression

### PR DESCRIPTION
**Pull request by zhengshiJ**
# Proposed changes

Issue Number: 

## Problem Summary:
scalar subquery:
A subquery that will return only one row and one column.

A limit has been added, where a = subquery returns a result of 1 row and 1 column. 
Here, the first limit is to return 1 column.

TODO: the subsequent return will limit the return to 1 row

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ScalarSubquery.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    public <R, C> R accept(ExpressionVisitor<R, C> visitor, C context) {
+        return visitor.visitScalarSubquery(this, context);
+    }
+
+    @Override
+    public Expression withChildren(List<Expression> children) {
+        Preconditions.checkArgument(children.size() == 1);
+        Preconditions.checkArgument(children.get(0) instanceof ScalarSubquery);
+        return new ScalarSubquery(((SubqueryExpr) children.get(0)).getQueryPlan());
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ScalarSubquery.java:62]: I think ScalarSubquery's children can be any plan, not only SubqueryExpr

Comment (924060929) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ScalarSubquery.java:62]: InSubquery's children also is any plan too.

Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ScalarSubquery.java]: done

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**